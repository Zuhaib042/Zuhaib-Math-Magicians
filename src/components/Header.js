import React from 'react';
import { Link } from 'react-router-dom';

export default function Header() {
  return (
    <>
      <header>
        <nav className="flex justify-between items-center border-b">
          <div className="py-2 ml-4">
            <span className="text-4xl  text-white">Math</span>
            <span className="text-4xl ml-1 tracking-wide text-[#EE4B2B]">
              Magician
            </span>
          </div>
          <ul className="sm:flex mr-4 text-white text-lg">
            <li className="px-2">
              <Link to="/">Home</Link>
            </li>
            <li className="px-2">
              <Link to="/calculator">Calculator</Link>
            </li>
            <li className="px-2">
              <Link to="/quote">Quote</Link>
            </li>
          </ul>
        </nav>
      </header>
    </>
  );
}
