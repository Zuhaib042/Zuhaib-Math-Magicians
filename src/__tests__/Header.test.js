import {  render, screen } from "@testing-library/react"
import Header from "../components/Header"

describe('Header', ()=>{
    it('should render header correctly', ()=>{
        render(<Header />)
        const homeTag = screen.getByText(/home/i);
        
        expect(homeTag).toMatchSnapshot()
     
    })
    it('should render header correctly', ()=>{
        render(<Header />)
        const calculatorTag = screen.getByRole('link', {name: 'Calculator'});
        
        expect(calculatorTag).toMatchSnapshot()
     
    })
})