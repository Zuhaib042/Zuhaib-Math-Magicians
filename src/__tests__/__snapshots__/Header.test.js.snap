// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header correctly 1`] = `
<a
  href="/"
>
  Home
</a>
`;

exports[`Header should render header correctly 2`] = `
<a
  href="/calculator"
>
  Calculator
</a>
`;
